parboiled-core/src/test/scala/org/parboiled2/BasicSpec.scala

```diff
@@ -17,188 +17,198 @@
 package org.parboiled2
 
 import shapeless.test.illTyped
-import org.specs2.specification.Scope
+import utest.{TestableString => _, _}
 
-class BasicSpec extends TestParserSpec {
 
-  "The Parser should correctly recognize/reject input for" >> {
+object BasicSpec extends TestParserSpec {
 
-    "simple char literals" in new TestParser0 {
-      def targetRule = rule { 'x' }
-      "x" must beMatched
-      "y" must beMismatched
-      "" must beMismatched
-    }
+  // TODO do we have a cleaner way to avoid the utest/parboiled '-' conflict?
```

Comment:
@sirthias this is the ugly part where I had to move the `rule/-` out of `Tests/-` territory to avoid conflict of the `-` symbol

Reply:
I see.

I think an easier solution that should be completely unintrusive to the tests would be adding this additional member to `TestParser`:

```scala
// shadow utests implicit extension on Strings which collides with our `str2CharRangeSupport`
val TestableString = null
```

This should do the trick.
If we get an `is never used` warning, we can "use" the `val` e.g. with

```scala
var input: ParserInput = TestableString
```

Reply:
Putting it in `TestParser` did not work, had to put it locally in the test and now it's indeed complaining about the unused val

Reply:
This works for me, without warnings:

```scala
abstract class TestParser[L <: HList, Out](implicit unpack: Unpack.Aux[L, Out]) extends Parser {
  // shadow utests implicit extension on Strings which collides with our `str2CharRangeSupport`
  def TestableString = null
  ...
```

and

```scala
"disallow compilation of an illegal character range" - new Parser {
  // shadow utests implicit extension on Strings which collides with our `str2CharRangeSupport`
  def TestableString = ParserInput.Empty
  def input = TestableString
  ...
```

Reply:
it all looks rather hacky but it works

Reply:
Yes, it's hacky.
But since it's only for making the tests work nicely and doesn't affect the library code itself I'd say it's acceptable.